### DISCUSSION TITLE
Zero downtime migration switching from redis to dragonflydb?

### DISCUSSION THREAD

**Discussion by phuongnd08** [dragonflydb/dragonfly#783] [Q&A]
Is it possible to achieve this? Like having dragondbfly replicate redis, then pause write to redis for a few seconds, waiting for dragondbfly to catch up completely and then start writing to dragondbfly?

**Comment by romange** [comment]
Yes, we support replication from Redis, but we have not yet exposed metrics that allow performing zero-time migration as you describe.  Do you use Lua scripts in production? What kind of frameworks do you use with Redis? The reason I am asking we still have some incompatibilities around Lua scripting that we are ironing out. If you want, send me "info all" stats from your redis node and I will tell you if it's safe to switch.

**Comment by phuongnd08** [comment]
We do use lua scripts. We use https://github.com/Shopify/wolverine

**Reply by romange** [reply]
In that case I suggest to test Dragonfly with your app in your test environment and see if it's working well. Currently Dragonfly requires that lua scripts will predeclare all the keys. If it's not the case it won't run EVAL.
